grid/language/named-entity-recognition/maps/ibm-cloud-natural-language-understanding.suma

```diff
@@ -0,0 +1,98 @@
+profile = "language/named-entity-recognition@1.0"
+provider = "ibm-cloud-natural-language-understanding"
+
+map NamedEntityRecognition { 
+  http POST "/instances/{parameters.INSTANCE_ID}/v1/analyze?version=2022-04-07" {
+    security "basic"
+  
+    request "application/json" {
+      body {
+        text = input.text
+        language = input.languageCode
+        features = {
+          "entities": {
+            "sentiment": false,
+            "mentions": false,
+            "emotion": false,
+          }
+        }
+      }
+    }
+
+    response 200 "application/json" {
+      entities = call foreach(entity of body.entities) MapEntity(
+        entity = entity
+      )
+
+      map result {
+        entities = entities
+      }
+    }
+
+    response 400 "application/json" {
+      map error {
+        title = "Unsuccessful Recognition",
+        detail = body.error,
+      }
+    }
+
+    response 401 "application/json" {
+      map error {
+        title = "Unauthorized",
+        detail = body.error,
+      }
+    }
+
+    response 403 "application/json" {
+      map error {
+        title = "Forbidden Resource",
+        detail = body.error,
+      }
+    }
+
+    response 500 {
+      map error {
+        title = "Problem on IBM's side",
+        detail = String(body),
+      }
+    }
+  }
+}
+
+operation MapEntity {
+  return {
+    text = args.entity.text,
+    category = call MapEntityCategory(ibmType = args.entity.type),
+    importance = args.entity.relevance,
+  }
+}
+
+
+operation MapEntityCategory {
```

Comment:
This is more for the profile: the mapping seems to be 1:1 with API. Have you looked at other APIs for a future unification?

Reply:
Yes, I also looked at some harmonized models (i.e. Eden) and they are all very similar, but use slightly different words (organization vs company etc). Then I wanted to use PascalCase values for the enum, and it turned out IBM has those. So, I decided not to change them for now, but prepared an explicit mapping mechanism for the future changes.